includes/admin/class-sensei-onboarding.php

```diff
@@ -114,6 +115,34 @@ public function admin_menu() {
 		}
 	}
 
+	/**
+	 * Redirect after first activation.
+	 */
+	public function activation_redirect() {
+		if (
+			// Check if activation redirect is needed.
+			! get_transient( 'sensei_activation_redirect' )
+			// Test whether the context of execution comes from async action scheduler.
+			// phpcs:ignore WordPress.Security.NonceVerification.Recommended -- Arguments used for comparison.
+			|| ( isset( $_REQUEST['action'] ) && 'as_async_request_queue_runner' === $_REQUEST['action'] )
+			// On these pages, or during these events, postpone the redirect.
+			|| wp_doing_ajax() || wp_doing_cron() || is_network_admin() || ! current_user_can( 'manage_sensei' )
```

Comment:
WooCommerce also doesn't do this when it detects a multi/bulk activation, I think. Do we also want to prevent that

https://github.com/woocommerce/woocommerce/blob/4f485a14deaedff4ba41fb95a1d50b180108563e/includes/admin/class-wc-admin.php#L163-L166

Reply:
Hum. Good catch! Added here: https://github.com/Automattic/sensei/pull/3102/commits/b07c07dac9322fbc69d89fc2637b34c58b39078a